### PR TITLE
fix: format behavior of union types

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 
re #627 

#### 2. What is the scope of this PR (e.g. component or file name):
fix the to string impl of union type:
- kclvm/ast/src/ast.rs

test cases:
kclvm/tools/src/format/test_data/format_data/union_types.golden
kclvm/tools/src/format/test_data/format_data/union_types.input
kclvm/tools/src/format/tests.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

fix: format behavior of union types

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/tools/src/format/test_data/format_data/union_types.golden [hunk 1]
```diff
@@ -2,4 +2,10 @@ workloadType: "Deployment" | "StatefulSet" = "Deployment"
 ServiceType: "LoadBalance" | "ClusterIP" = "LoadBalance"
 abc: A | B | C = A {}
 
+schema Server():
+    [name: str | int]: str | int
+    workloadType: "Deployment" | "StatefulSet"
+    dict: {str:str | int}
+
+type some_alias = "abc" | "def" | {str | int:}
```
Comment (Peefy) [inline kclvm/tools/src/format/test_data/format_data/union_types.golden:10]: s/{str | int:}/{str: int}

File: kclvm/tools/src/format/test_data/format_data/union_types.input [hunk 2]
```diff
@@ -2,4 +2,10 @@ workloadType: "Deployment"|"StatefulSet" = "Deployment"
 ServiceType: "LoadBalance"   |"ClusterIP" = "LoadBalance"
 abc: A|B|C = A{}
 
+schema Server():
+    [name: str|int]: str|int
+    workloadType: "Deployment"|    "StatefulSet"
+    dict: {str: str|int}
+
+type some_alias = "abc"|"def"|{str|int:}
```
Comment (Peefy) [inline kclvm/tools/src/format/test_data/format_data/union_types.input:10]: s/{str|int:}/{str:int}

File: kclvm/ast/src/ast.rs [hunk 3]
```diff
@@ -574,7 +574,7 @@ pub struct SchemaIndexSignature {
 pub struct SchemaAttr {
     pub doc: String,
     pub name: NodeRef<String>,
-    pub type_str: NodeRef<String>,
+pub type_str: NodeRef<String>,
```
Comment (Peefy) [inline kclvm/ast/src/ast.rs:577]: Format the code.

Comment (amyXia1994) [reply kclvm/ast/src/ast.rs]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**